src/main/java/com/google/sps/data/BusinessesService.java

```diff
@@ -33,10 +41,43 @@
   /** Create a new Businesses instance
   * @param businesses businesses from SmallCityService
   **/
+
   public BusinessesService(List<Listing> businesses) {
     this.businesses = businesses;
   }
 
+  public PreparedQuery connectToBigBusinessDatabase(){
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    Query query = new Query("BigBusinesses");
+    PreparedQuery dateabaseResults = datastore.prepare(query);
+    return dateabaseResults;
+  }
+
+  public List<Listing> removeBigBusinessesFromResults(PreparedQuery dateabaseResults){
+    Iterator<Listing> businessesList =  businesses.iterator();
+    Entity entity;
+    Boolean untilBigBusinessIsFoundInTheBusinessesList = true;
+    String businessName;
+    while (businessesList.hasNext()) {
+      Listing currentListing = businessesList.next();
+      Iterator<Entity> bigBusinessesEntities =  dateabaseResults.asIterable().iterator();
+      while(bigBusinessesEntities.hasNext() && untilBigBusinessIsFoundInTheBusinessesList) {
```

Comment:
nit: You can shorten "untilBigBusinessIsFoundInTheBusinessesList" to "bigBusinessIsFound" without losing necessary context.

Reply:
I fixed this, but changed this to untilBigBusinessIsFound, because I set it to true initially